src/sources/grib.jl

```diff
@@ -0,0 +1,165 @@
+const GRIB_DIM_MAP = Dict(
+    "latitude" => Y,
+    "longitude" => X,
+    "valid_time" => Ti,
+    "level" => Z,
+    "vertical" => Z,
+    "x" => X,
+    "y" => Y,
+    "z" => Z,
+)
+
+haslayers(::Type{GRIBfile}) = true
+defaultcrs(::Type{GRIBfile}) = EPSG(4326)
+defaultmappedcrs(::Type{GRIBfile}) = EPSG(4326)
+
+# Implement some methods on Dataset and Variable that are needed for conversion to Raster
+Base.Array(var::GDS.Variable) = Array(var.values)
+DA.readblock!(var::GDS.Variable, args...) = DA.readblock!(var.values, args...)
+
+# Raster ########################################################################
+
+function Raster(ds::GDS.Dataset, filename::AbstractString, key=nothing; kw...)
+    key = _firstkey(ds, key)
+    Raster(ds[key], filename, key; kw...)
+end
+
+_firstkey(ds::GDS.Dataset, key::Nothing=nothing) = Symbol(first(layerkeys(ds)))
+_firstkey(ds::GDS.Dataset, key) = Symbol(key)
+
+_dimkeys(ds::GDS.Dataset) = collect(keys(ds.dims))
+_dimnames(var::GDS.Variable) = collect(keys(var.dims))
+
+
+function FileArray(var::GDS.Variable, filename::AbstractString; kw...)
+    da = RasterDiskArray{GRIBfile}(var.values)
+    size_ = size(da)
+    eachchunk = DA.eachchunk(da)
+    haschunks = DA.haschunks(da)
+    T = eltype(var)
+    N = length(size_)
+    FileArray{GRIBfile,T,N}(filename, size_; eachchunk, haschunks, kw...)
+end
+
+
+# DimensionalData methods for CfGRIB.Dataset types ###############################
+
+function DD.dims(ds::GDS.Dataset, crs=nothing, mappedcrs=nothing)
+    map(_dimkeys(ds)) do key
+        _dsdim(ds, key, crs, mappedcrs)
+    end |> Tuple
+end
+function DD.dims(var::GDS.Variable, crs=nothing, mappedcrs=nothing)
+    names = _dimnames(var)
+    ds = var.ds
+    map(names) do name
+        _dsdim(ds, name, crs, mappedcrs)
+    end |> Tuple
+end
+
+DD.metadata(ds::GDS.Dataset) = _metadatadict(GRIBfile, ds.attrib)
+DD.metadata(var::GDS.Variable) = _metadatadict(GRIBfile, var.attrib)
+
+function DD.layerdims(ds::GDS.Dataset)
+    keys = Tuple(layerkeys(ds))
+    dimtypes = map(keys) do key
+        DD.layerdims(ds[string(key)])
+    end
+    NamedTuple{map(Symbol, keys)}(dimtypes)
+end
+function DD.layerdims(var::GDS.Variable)
+    map(_dimnames(var)) do dimname
+        _gdsdimtype(dimname)()
+    end |> Tuple
+end
+
+DD.layermetadata(ds::GDS.Dataset) = _layermetadata(ds, Tuple(layerkeys(ds)))
+function _layermetadata(ds::GDS.Dataset, keys)
+    dimtypes = map(k -> DD.metadata(ds[string(k)]), keys)
+    NamedTuple{map(Symbol, keys)}(dimtypes)
+end
+
+missingval(var::GDS.Variable) = missing
```

Comment:
Are you actually converting nodata values to `missing` in GRIBDatasets.jl ?

Reply:
Yes, I am

Reply:
I meant the commented out bits